frontend/src/Hooks/useForm.js

```diff
@@ -0,0 +1,26 @@
+import { useState } from "react";
+
+const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+const isInvalid = (key, newValue) => {
+  return key === "email" ? !emailRegex.test(newValue) : newValue === "";
+};
+
+const useForm = items => {
+  const [formState, setFormState] = useState(items);
+
+  return [
+    formState,
+    (key, newValue) =>
```

Comment:
You can simplify this even further. Make it accept one argument, which will be the `event` from handler. Then you can set the propery like `[event.target.name]: { invalid, touched, value: event.target.value }`. And then you can very simply pass it to the onChange prop like this: `onChange={formSetter}`